### PR TITLE
Setup Travis ci

### PR DESCRIPTION
**Pull request by skabashnyuk**
On each commit
 - run test
- build and push docker image ```ksmster/jwtproxy:latest``` ```ksmster/jwtproxy:branch-name```,```ksmster/jwtproxy:commitid```,```ksmster/jwtproxy:build-number```

On each tag
 - run test
- create release on github
- build and push docker image ```ksmster/jwtproxy:latest```  and ```ksmster/jwtproxy:tag_name```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+- docker pull $REPO:latest
+- docker build --cache-from $REPO:latest -f Dockerfile -t  $REPO:latest .
+- docker push $REPO
+deploy:
+  skip_cleanup: true
+  provider: releases
+  api_key:
+    secure: 
+  file:
+  - kubsrv_Linux_x86_64
```
Comment (sleshchenko) [inline .travis.yml]: I guess it is copy/paste error from `kubsrv` that is REST API for Feature API. Did I miss something?

Comment (skabashnyuk) [reply .travis.yml]: no. you are right. will fix it.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  directories:
+  - "$GOPATH/pkg/dep"
+env: 
+before_deploy:
+- CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-w -s' -a -installsuffix cgo -o jwtproxy
+  gucmd/jwtproxy/main.go
+- echo $DOCKER_PASS | docker login -u $DOCKER_USER --password-stdin
+- export REPO=ksmster/jwtproxy
+- export TAG=`if [ "$TRAVIS_BRANCH" == "master" ]; then echo "latest"; else echo $TRAVIS_BRANCH
+  ; fi`
```
Comment (sleshchenko) [inline .travis.yml]: it would be better to move it to line before

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+- docker build --cache-from $REPO:latest -f Dockerfile -t  $REPO:$TAG .
+- docker tag $REPO:$TAG $REPO:latest
+- docker push $REPO
+after_success:
+- CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-w -s' -a -installsuffix cgo -o jwtproxy
+  cmd/jwtproxy/main.go
+- echo $DOCKER_PASS | docker login -u $DOCKER_USER --password-stdin
+- export REPO=ksmster/jwtproxy
+- export TAG=`if [ "$TRAVIS_BRANCH" == "master" ]; then echo "latest"; else echo $TRAVIS_BRANCH
+  ; fi`
```
Comment (sleshchenko) [inline .travis.yml]: it would be better to move it to line before

**Pull request merged**